datafusion/core/src/dataframe/mod.rs

```diff
@@ -522,6 +522,38 @@ impl DataFrame {
         })
     }
 
+    /// Return a new `DataFrame` with duplicated rows removed as per the specified expression list
+    /// according to the provided sorting expressions grouped by the `DISTINCT ON` clause
+    /// expressions.
+    ///
+    /// # Example
+    /// ```
+    /// # use datafusion::prelude::*;
+    /// # use datafusion::error::Result;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// let ctx = SessionContext::new();
+    /// let df = ctx.read_csv("tests/data/example.csv", CsvReadOptions::new()).await?
+    ///   // Return a single row (a, b) for each distinct value of a  
```

Comment:
Wondering if DISTINCT supports all types, incl binary, complex, etc? if its not we should be mentioning it in the doc, and double check it returns a respective error instead of crash/corruption

Reply:
It is a good idea -- I filed https://github.com/apache/datafusion/issues/11052 to track 

However, I think it is not required for this PR so I merged it in